src/util.rs

```diff
@@ -25,3 +27,26 @@ pub fn l2_normalize_array(mut v: ArrayViewMut2<f32>) -> Array1<f32> {
 
     norms.into()
 }
+
+pub fn read_number(reader: &mut BufRead, delim: u8) -> Result<usize> {
```

Comment:
`read_number` taking a `delim` argument is somewhat counterintuitive. Maybe add rustdocs to clear this up?

Reply:
Note that this is only `pub(crate)`, but I can do a separate PR adding rustdoc.

Reply:
That one always gets me in PRs. I always assume `pub fn` means public interface. For maintainers it's still nice to have, I guess.

Reply:
>That one always gets me in PRs.

Me too. But `pub(crate)`ing the whole module prevents accidental inclusion. Would be nice if GitHub could include automatic annotations (actual visibility).

Reply:
Something like this: https://drewdevault.com/2019/07/08/Announcing-annotations-for-sourcehut.html

One can always hope ;).